README.md

```diff
@@ -19,52 +19,65 @@ YourAppDelegate.h
 
 YourAppDelegate.m
 ```objective-c
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+    EDQueueStorageEngine *fmdbBasedStorage = [[EDQueueStorageEngine alloc] initWithName:@"database.sample.sqlite"];
+
+    self.persistentTaskQueue = [[EDQueue alloc] initWithPersistentStore:fmdbBasedStorage];
+
+    /* There you go ^^ */
+
+    return YES;
+}
+
+//
+
 - (void)applicationDidBecomeActive:(UIApplication *)application
 {
-    [[EDQueue sharedInstance] setDelegate:self];
-    [[EDQueue sharedInstance] start];
+    [self.persistentTaskQueue setDelegate:self];
+    [self.persistentTaskQueue start];
 }
 
 - (void)applicationWillResignActive:(UIApplication *)application
 {
-    [[EDQueue sharedInstance] stop];
+    [self.persistentTaskQueue stop];
 }
 
-- (EDQueueResult)queue:(EDQueue *)queue processJob:(NSDictionary *)job
+- (void)queue:(EDQueue *)queue processJob:(EDQueueJob *)job completion:(void (^)(EDQueueResult))block
 {
-    sleep(1);           // This won't block the main thread. Yay!
+    sleep(1);
     
-    // Wrap your job processing in a try-catch. Always use protection!
     @try {
-        if ([[job objectForKey:@"task"] isEqualToString:@"success"]) {
-            return EDQueueResultSuccess;
-        } else if ([[job objectForKey:@"task"] isEqualToString:@"fail"]) {
-            return EDQueueResultFail;
+        if ([job.task isEqualToString:@"success"]) {
+            block(EDQueueResultSuccess);
+        } else if ([job.task isEqualToString:@"fail"]) {
+            block(EDQueueResultFail);
+        } else {
+            block(EDQueueResultCritical);
         }
     }
     @catch (NSException *exception) {
-        return EDQueueResultCritical;
+        block(EDQueueResultCritical);
     }
-    
-    return EDQueueResultCritical;
 }
 ```
 
 SomewhereElse.m
 ```objective-c
-[[EDQueue sharedInstance] enqueueWithData:@{ @"foo" : @"bar" } forTask:@"nyancat"];
+[self.queue enqueueWithData:@{ @"foo" : @"bar" } forTask:@"nyancat"];
 ```
 
-In order to keep things simple, the delegate method expects a return type of `EDQueueResult` which permits three distinct states:
+Explanation to type of `EDQueueResult` which permits three distinct states:
```

Comment:
Yepp, let's revert this one. But the point Is I've removed synchronous delegate at all. Therefore - it is always one entry point, always async.